Entities/MOPixel.cpp

```diff
@@ -1,450 +1,255 @@
+
+	void MOPixel::SetAtom(Atom *newAtom) {
+		delete m_Atom;
+		m_Atom = newAtom;
+		m_Atom->SetOwner(this);
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void MOPixel::SetLethalRange(float range) {
+		m_LethalRange = range;
+		if (m_MinLethalRange < m_MaxLethalRange) { m_LethalRange *= RangeRand(m_MinLethalRange, m_MaxLethalRange); }
+
+		// convert to meters
+		m_LethalRange /= c_PPM;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void MOPixel::Travel() {
+		MovableObject::Travel();
+
+		if (m_PinStrength) {
+			return;
+		}
+
+		// Set the atom to ignore a certain MO, if set and applicable.
+		if (m_HitsMOs && m_pMOToNotHit && g_MovableMan.ValidMO(m_pMOToNotHit) && !m_MOIgnoreTimer.IsPastSimTimeLimit()) {
+			MOID root = m_pMOToNotHit->GetID();
+			int footprint = m_pMOToNotHit->GetMOIDFootprint();
+			for (int i = 0; i < footprint; ++i) {
+				m_Atom->AddMOIDToIgnore(root + i);
+			}
+		}
+		// Do static particle bounce calculations.
+		int hitCount = m_Atom->Travel(g_TimerMan.GetDeltaTimeSecs(), true, g_SceneMan.SceneIsLocked());
+
+		// Now clear out the ignore override for next frame
+		m_Atom->ClearMOIDIgnoreList();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool MOPixel::CollideAtPoint(HitData &hd) {
+		RTEAssert(hd.HitPoint.GetFloored() == m_Pos.GetFloored(), "Collision mismatch in MOPixel::CollideAtPoint!");
+		RTEAssert(hd.Body[HITOR], "Valid MO not passed into MOPixel::CollideAtPoint!");
+
+		hd.TotalMass[HITEE] = m_Mass;
+
+		// See if we were already hit by this MO earlier during this frame update.
+		if (m_AlreadyHitBy.find(hd.Body[HITOR]->GetID()) != m_AlreadyHitBy.end()) {
+			// TODO: Find out so that if we what.
+			// If we were hit, then remove so that if we 
+			m_AlreadyHitBy.erase(hd.Body[HITOR]->GetID());
+		} else {
+			m_AlreadyHitBy.insert(hd.Body[HITOR]->GetID());
+		}
+		return true;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void MOPixel::RestDetection() {
+		MovableObject::RestDetection();
+
+		// If we seem to be about to settle, make sure we're not still flying in the air
+		if ((m_ToSettle || IsAtRest()) && g_SceneMan.OverAltitude(m_Pos, 2, 0)) {
+			m_RestTimer.Reset();
+			m_ToSettle = false;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void MOPixel::Update() {
+		MovableObject::Update();
+
+		if (m_HitsMOs && m_Sharpness > 0) {
+			m_DistanceTraveled += m_Vel.GetLargest() * g_TimerMan.GetDeltaTimeSecs();
+			if (m_DistanceTraveled > m_LethalRange) {
+				if (m_Sharpness < m_LethalSharpness) {
```

Comment:
I dunno this isn't my code and I didn't bork it in the process.
```c++
    else if (m_HitsMOs && m_Sharpness > 0)
    {
        m_DistanceTraveled += m_Vel.GetLargest() * g_TimerMan.GetDeltaTimeSecs();
        if (m_DistanceTraveled > m_LethalRange)
        {
			
            if (m_Sharpness < m_LethalSharpness)
                m_Sharpness = max(m_Sharpness * (1.0 - (20.0 * g_TimerMan.GetDeltaTimeSecs())) - 0.1, 0.0);
            else
                m_Sharpness *= 1.0 - (10.0 * g_TimerMan.GetDeltaTimeSecs());
			
			if (m_LethalRange > 0)
				m_HitsMOs = false;
        }
    }
```
IIRC it works but I only ever used it once.

Reply:
I'm 100% sure all the sharpness stuff here doesn't work, when range > lethal range it goes HitsMOs = false.

Aside from that I can see what's going on now, but it doesn't match up with the member variable comment for LethalSharpness. Basically what's happening is (ignoring the hitsmos = false part) is, once it gets past lethal range
1. If sharpness >= lethal sharpness, decrease it at some rate
2. If sharpness < lethal sharpness, decrease at some other rate until it gets to 0

So yeah, none of this matches what's suggested. I seem to recall weegee first put sharpness decreasing at range and then replaced it with just making bullets not hit mos, so that's probably what we're seeing, a bit of mess of the 2 systems being mixed up.

Reply:
I'm not planning deal with this so i'm just going to slap a todo on it.